actk/steps/make_diagnostic_sheet/make_diagnostic_sheet.py

```diff
@@ -0,0 +1,487 @@
+                        dataset,
+                        this_figure_number,
+                        sublot,
+                        index,
+                        this_metadata,
+                        metadata,
+                        multiple_paths,
+                        split_fig_path,
+                        feature,
+                        fig_width,
+                        fig_height,
+                    )
+
+            log.info(
+                f"Completed diagnostic sheet for: {this_metadata} {this_figure_number} "
+            )
+
+        return dataset
+
+    @staticmethod
+    def _collect_group(
+        row_index: int,
+        row: pd.Series,
+        diagnostic_sheet_dir: Path,
+        overwrite: bool,
+        metadata: str,
+    ) -> Union[DiagnosticSheetResult, DiagnosticSheetError]:
+        # Don't use dask for image reading
+        aicsimageio.use_dask(False)
+
+        try:
+            # Get the ultimate end save paths for grouped plot
+            if row[str(metadata)] or row[str(metadata)] == 0:
+                assert DatasetFields.CellImage2DAllProjectionsPath in row.index
+                save_path = (
+                    diagnostic_sheet_dir / f"{metadata}_{row[str(metadata)]}.png"
+                )
+                log.info(
+                    f"Generating diagnostic sheet path for cell ID: {row.CellId},"
+                    f"{metadata} {row[str(metadata)]}"
+                )
+            else:
+                # else no path to save
+                save_path = None
+                # diagnostic_sheet_save_path = (
+                #     diagnostic_sheet_dir / f"{metadata}_{row[str(metadata)]}.png"
+                # )
+
+            # Check skip
+            if not overwrite and save_path.is_file():
+                log.info(
+                    f"Skipping diagnostic sheet path for cell ID: {row.CellId},"
+                    f"{metadata} {row[str(metadata)]}"
+                )
+                return DiagnosticSheetResult(row.CellId, None)
+
+            # Return ready to save image
+            return DiagnosticSheetResult(row.CellId, str(save_path))
+        # Catch and return error
+        except Exception as e:
+            log.info(
+                f"Failed to retrieve the CellImage2DAllProjectionsPath"
+                f"for cell ID: {row.CellId},"
+                f"{metadata} {row[str(metadata)]}"
+                f"Error: {e}"
+            )
+            return DiagnosticSheetError(row.CellId, str(e))
+
+    @log_run_params
+    def run(
+        self,
+        dataset: Union[str, Path, pd.DataFrame, dd.DataFrame],
+        max_cells: int = 1000,
+        distributed_executor_address: Optional[str] = None,
+        batch_size: Optional[int] = None,
+        overwrite: bool = False,
+        feature: Optional[str] = None,
+        metadata: Optional[Union[list, str]] = None,
+        fig_width: Optional[int] = None,
+        fig_height: Optional[int] = None,
```

Comment:
I would move these to above the `distributed_executor_address`. It's nitpicky. But these parameters are for the step itself while `distributed_executor_address`, `batch_size`, and `overwrite` are from the flow. If that makes sense?